hashentry.go

```diff
@@ -176,6 +176,16 @@ func (h *HashEntry) codify(v string) error {
 	return nil
 }
 
+// AlgIDToString provides a conversion from the algorithm ID to the string
+// representation of the algorithm
+func (h *HashEntry) AlgIDToString() (string, error) {
+	sAlg, ok := algToString[h.HashAlgID]
+	if !ok {
+		return "", fmt.Errorf("unknown hash algorithm ID %d", h.HashAlgID)
```

Comment:
why is this an error?  we could instead return something like `"alg-id(%d)"`

Reply:
It is an error because the algorithm ID doesn't match to the ones defined in the list of valid ones above (based on the IANA format), so it is unknown/

Reply:
sure, but the [IANA registry](https://www.iana.org/assignments/named-information/named-information.xhtml) is not cast in stone: new algorithms may be registered in the future - and may have been registered since we first implemented this.

Reply:
Ok that makes sense, so we want the return type to be (string, string) instead of (string, error)?

Reply:
I think it'd be `string` only:
```suggestion
		return fmt.Sprint("alg-id(%d)", h.HashAlgID)
```